File: DownBall/Assets/Scripts/MovingObstacles.cs
using UnityEngine;
using System.Collections;

public class MovingObstacles : MonoBehaviour {

    public Vector3 pointB;
     
    IEnumerator Start () {
        Vector3 pointA = transform.position;
        while (true) {
            yield return MoveObject(pointA, pointB, 3);
            yield return MoveObject(pointB, pointA, 3);
        }
    }
     
    IEnumerator MoveObject (Vector3 startPos, Vector3 endPos, float time) {
        float i = 0.0f;
        float rate = 1.0f / time;
        while (i < 1.0f) {
            i += Time.deltaTime * rate;
            transform.position = Vector3.Lerp(startPos, endPos, i);
            yield break; 
        }
    }
}

File: DownBall/Assets/Scripts/MovingObstacles.js
#pragma strict

var pointB : Vector3;
var degrees : int;
 
function Start () {
    var pointA = transform.position;
    while (true) {
        yield MoveObject(transform, pointA, pointB, 3.0);
        yield MoveObject(transform, pointB, pointA, 3.0);
    }
}
 
function MoveObject (thisTransform : Transform, startPos : Vector3, endPos : Vector3, time : float) {
    var i = 0.0;
    var rate = 1.0/time;
    
    while (i < 1.0) {
        i += Time.deltaTime * rate;
        thisTransform.position = Vector3.Lerp(startPos, endPos, i);
        transform.Rotate (0,0,degrees*Time.deltaTime);
        yield; 
    }
}